app/src/main/assets/native-10.7/ExoPlayerPlugin.js

```diff
@@ -1,3 +1,5 @@
+import qualityOptions from 'components/qualityOptions';
```

Comment:
I'm quite sure this import is not going to work since jf-web is bundled. We probably need to make a change in jf-web to add the qualityOptions to the plugin constructor.

Reply:
Do we need the helper functions from this component or can we re-implement it in Kotlin?

Reply:
> I'm quite sure this import is not going to work since jf-web is bundled. We probably need to make a change in jf-web to add the qualityOptions to the plugin constructor.

Very likely, but at least in 10.6.4 it works without the need to make changes in jf-web using "define", I don't know what the equivalent is since they converted the AMD module to ES6.

> Do we need the helper functions from this component or can we re-implement it in Kotlin?

Well, it's possible, but then the existing code would be duplicated and it would have to be maintained so that it's in sync with jf-web in case of changes.

https://github.com/jellyfin/jellyfin-web/blob/master/src/components/qualityOptions.js

Reply:
Since we're aiming for fully native we need to bring it to the native side eventually anyway. Besides, this PR will likely end up in the release that also drops 10.6 support.

Reply:
Got it, just to confirm, as this differs from what I had read before, and I agree with @ferferga comment, did they come up with something or is it okay to separate this for now? Maybe we are still on time, I don't know, maybe an endpoint can be added to obtain the qualities before the next rc / stable version of Jellyfin 10.7?

See https://github.com/jellyfin/jellyfin-web/pull/1993#issuecomment-723371174

Reply:
The discussion continues here: https://github.com/jellyfin/jellyfin/discussions/4795 but it definitely won't happen in 10.7. for now we should implement it client side in the kotlin code

Reply:
In fact I opened it after getting the mention here, so thanks @CarlosOlivo for reminding me! ^^

As @nielsvanvelzen points out, this involves a massive amount of changes server-side that probably won't happen until server guys are happy with the code there (and there's so much work to do there to have a clean codebase).

However, everything that can be done up to this day to set the best foundation possible for that hypothetic system proposal it's great!